Add type-checking to unplugin (#689)
DanielXMoore/Civet: integration/unplugin/README.md

  dts?: boolean;
  outputExtension?: string;
  js?: boolean;
  typecheck?: boolean;
  transformOutput?: (
    code: string,
    id: string
  ) => TransformResult | Promise<TransformResult>;
}
```

> - `dts`: This also supports generating `.d.ts` type definition files from the civet source, which is useful for building libraries. Default: `false`

Reviewer: ```suggestion
- `dts`: Whether to generate `.d.ts` type definition files from the Civet source, which is useful for building libraries. Default: `false`
```
